### PR TITLE
Tests for new nets fields

### PR DESCRIPTION
**Pull request by fasaxc**
## Description
Add tests for the multiple CIDRs in a rule feature.

- [x] Split out the profile tests into individual methods.
- [x] Add one that covers multiple CIDRs.

## Todos
- [x] Tests
- [x] Documentation

### CODE REVIEW

File: master/reference/calicoctl/resources/policy.md [hunk 1]
```diff
@@ -92,8 +92,10 @@ applied action is deny.
 |-------------|-----------------------------|-------------------|--------|------------|
 | tag (deprecated)      | Match on tag. |  | string | |
 | notTag (deprecated)   | Negative match on tag. |  | string | |
-| net    | Match on CIDR. | Valid IPv4 or IPv6 CIDR  | cidr | |
-| notNet | Negative match on CIDR. | Valid IPv4 or IPv6 CIDR | cidr | |
+| net                   | Deprecated (use "nets" instead): Match on CIDR. | Valid IPv4 or IPv6 CIDR  | cidr | |
```
Comment (neiljerram) [inline master/reference/calicoctl/resources/policy.md]: Do we have to deprecate the singular forms?  I would guess (before looking at the code changes) that the cost of this is very small, so why not keep both singular and plural forms on equal footing?

Comment (neiljerram) [inline master/reference/calicoctl/resources/policy.md]: OTOH, if we keep this deprecation, let's put the non-deprecated forms first.

File: master/reference/calicoctl/resources/policy.md [hunk 2]
```diff
@@ -92,8 +92,10 @@ applied action is deny.
...
+| nets                  | Match any one of the listed CIDRs. | List of valid IPv4 or IPv6 CIDRs  | list of cidr |
+| notNet                | Deprecated (use "notNets" instead): Negative match on CIDR. | Valid IPv4 or IPv6 CIDR | cidr | |
+| nets                  | Negative match on CIDRs. | List of valid IPv4 or IPv6 CIDRs  | list of cidr |
```
Comment (neiljerram) [inline master/reference/calicoctl/resources/policy.md]: 1. Typo: notNets
2. "Match IPs that are not in any of the listed CIDRs"
3. "List of invalid ..."

File: master/reference/calicoctl/resources/profile.md [hunk 3]
```diff
@@ -80,8 +81,10 @@ spec:
 |-------------|-----------------------------|-------------------|--------|------------|
 | tag (deprecated)      | Positive match on tag. |  | string | |
 | notTag (deprecated)   | Negative match on tag. |  | string | |
-| net    | Match on CIDR. | Valid IPv4 or IPv6 CIDR  | cidr | |
-| notNet | Negative match on CIDR. | Valid IPv4 or IPv6 CIDR | cidr | |
+| net                   | Deprecated (use "nets" instead): Match on CIDR. | Valid IPv4 or IPv6 CIDR  | cidr | |
+| nets                  | Match any one of the listed CIDRs. | List of valid IPv4 or IPv6 CIDRs  | list of cidr |
+| notNet                | Deprecated (use "notNets" instead): Negative match on CIDR. | Valid IPv4 or IPv6 CIDR | cidr | |
+| nets                  | Negative match on CIDRs. | List of valid IPv4 or IPv6 CIDRs  | list of cidr |
```
Comment (neiljerram) [inline master/reference/calicoctl/resources/profile.md]: Same points as in master/reference/calicoctl/resources/policy.md

File: master/reference/calicoctl/resources/policy.md (hunk 2 above)
Comment (fasaxc) [inline master/reference/calicoctl/resources/policy.md]: Agree with 1 & 2 but not sure about 3; the list still needs to contain valid CIDRs.

Comment (neiljerram) [inline master/reference/calicoctl/resources/policy.md]: Ah, some subtlety here: I meant "invalid" as in "these are the CIDRs that I want to disallow"; you mean "invalid" as in syntactically invalid.  I think it's tricky to get both of those concepts clearly into one sentence; perhaps "List of valid IPv4 or IPv6 CIDRs to disallow"?

On the other hand, if it's clear that the 3rd column is only describing syntax, and not function, perhaps it was all OK as it was before.

File: calico_node/Makefile [hunk 4]
```diff
@@ -1,6 +1,6 @@
 ###############################################################################
 # Versions:
-RELEASE_STREAM?=v2.3
+RELEASE_STREAM?=master
```
Comment (caseydavenport) [inline calico_node/Makefile]: Is this just left over or do we actually want this change?

Comment (fasaxc) [inline calico_node/Makefile]: Might not be needed after your change but does it make sense for `make st` to run with `v2.3` versions?  I'd kinda expect that to use the current code/master.

File: calico_node/tests/st/policy/test_profile.py [hunk 5]
```diff
@@ -27,200 +27,277 @@
...
+                 'destination': {'nets': n1_ips[0:1],
+                                 'notNets': n1_ips[1:]}}
+        rule2 = {'action': 'allow',
+                 'destination': {'nets': n1_ips[1:2],
+                                 'notNets': n1_ips[:1]}}
+        prof_n1['spec']['egress'] = [rule1, rule2]
+        self._apply_new_profile(self.new_profiles, self.host1)
+        self.assert_connectivity(retries=2,
+                                 pass_list=self.n1_workloads[:2],
+                                 fail_list=self.n1_workloads[2:])
```
Comment (lwr20) [inline calico_node/tests/st/policy/test_profile.py]: Tests we might like to add: 
Rules with: 
- CIDRs other than /32
- the same CIDR twice
- overlapping CIDRs

File: calico_node/Makefile (hunk 4 above)
Comment (caseydavenport) [inline calico_node/Makefile]: > Update the RELEASE_STREAM variable at the top of calico_node/Makefile to the new major/minor version.

I think the weird bit is that RELEASE_STREAM is currently updated as part of the release process, so changing this to master doesn't really jive with that.

I think we need to revert this change and potentially change it in the near future, but only after thinking about impacts to our release process.

In the meantime, `make RELEASE_STREAM=master st` will work (though it's pretty annoying).

**Pull request merged**